src/yz_kv.erl

```diff
@@ -190,6 +190,48 @@ has_indexes(RemoteNode) ->
         _ -> false
     end.
 
+
+%% @doc Index the data supplied in the Riak Object.
+%% The Riak Object should be a deserialized object (suitable for use with the riak_object
+%% module, or a binary, which has been serialized using riak_object:to_binary/1
+-spec index_binary(bucket(), key(), binary(), atom(), integer()) -> ok.
+index_binary(B, K, Obj, Reason, P) when is_binary(Obj) ->
+    BKey = {B, K},
+    case yokozuna:is_enabled(index) andalso ?YZ_ENABLED of
+        true ->
+            Ring = yz_misc:get_ring(transformed),
+            case is_owner_or_future_owner(P, node(), Ring) of
+                true ->
+                    T1 = os:timestamp(),
+                    try
```

Comment:
@fadushin thinking we can create a helper to avoid the duplication in index/3 and index/5, no?